t/api/14-plugin_obs_rsync.t

```diff
@@ -41,29 +47,67 @@ my $app = $t->app;
+    $t->put_ok('/api/v1/obs_rsync/Proj2/runs')
+      ->status_is(201, "Proj2 first time - should just start as queue is empty");
+    $t->put_ok('/api/v1/obs_rsync/Proj2/runs')
+      ->status_is(208, "Proj2 second time - should report IN_QUEUE, because another Proj2 wasn't started by worker");
+    $t->put_ok('/api/v1/obs_rsync/Proj3/runs')->status_is(201, "Proj3 first time - should just start");
+    $t->put_ok('/api/v1/obs_rsync/Proj2/runs')->status_is(208, "Proj2 still gets queued");
+    $t->put_ok('/api/v1/obs_rsync/Proj3/runs')->status_is(208, "Proj3 now reports that already queued");
+    $t->put_ok('/api/v1/obs_rsync/Proj1/runs')
+      ->status_is(507, "Proj1 cannot be handled because queue is full 2=(Proj2, Proj3 running)");
+    $t->put_ok('/api/v1/obs_rsync/Proj3/runs')->status_is(208, "Proj3 is still in queue");
+    $t->put_ok('/api/v1/obs_rsync/WRONGPROJECT/runs')->status_is(404, "wrong project still returns error");
 
-$t->get_ok('/admin/obs_rsync/Leap:15.1:ToTest/runs')->status_is(200, "project logs status")
-  ->element_exists('[.run_190703_143010]');
+    $t->app->start('gru', 'run', '--oneshot');
 
-$t->get_ok('/admin/obs_rsync/Leap:15.1:ToTest/runs/.run_190703_143010')->status_is(200, "project log subfolder status")
-  ->element_exists('[files_iso.lst]');
+    $t->put_ok('/api/v1/obs_rsync/Proj1/runs')->status_is(201, "Proj1 just starts as queue is empty now");
+}
 
-$t->get_ok('/admin/obs_rsync/Leap:15.1:ToTest/runs/.run_190703_143010/download/files_iso.lst')
-  ->status_is(200, "project log file download status")
-  ->content_like(qr/openSUSE-Leap-15.1-DVD-x86_64-Build470.1-Media.iso/)
-  ->content_like(qr/openSUSE-Leap-15.1-NET-x86_64-Build470.1-Media.iso/);
+subtest 'test queue' => sub {
+    test_queue($t);
+};
+$t->app->start('gru', 'run', '--oneshot');
 
-$t->reset_session;
+subtest 'test queue again' => sub {
+    test_queue($t);
+};
+$t->app->start('gru', 'run', '--oneshot');
 
-$t->put_ok('/api/v1/obs_rsync/Leap:15.1:ToTest/runs')->status_is(201, "trigger rsync");
-$t->put_ok('/api/v1/obs_rsync/WRONGPROJECT/runs')->status_is(404, "trigger rsync wrong project");
-$t->put_ok('/admin/obs_rsync/Leap:15.1:ToTest/runs')->status_is(404, "trigger rsync non-api path");
+# ObsRsync::Task#_run are called from Gru worker, so it is covered
+# but let's test it explicitly as well to make codecov happy
+use OpenQA::WebAPI::Plugin::ObsRsync::Task;
+
+subtest 'make codecov happy' => sub {
+    {
+        package Test::FakeMinionJob;
+        sub note {
+        }
+        sub retry {
+        }
+        sub info {
+        }
+        sub finish {
+            return 222;
+        }
+        sub fail {
+            Test::More::fail("Minion job shouldn't have failed.");
+            Test::More::note(Data::Dumper::Dumper(\@_));
+        }
+    }
+    my %args = (project => 'Proj1');
+    my $res  = OpenQA::WebAPI::Plugin::ObsRsync::Task::run($t->app, bless({} => 'Test::FakeMinionJob'), \%args);
+    ok($res = 222);
```

Comment:
This test makes no sense.

Reply:
Yes, I would just get rid of that test if it is covered anyways (even if codecov fails to take that coverage into account because it is inside a GRU task).